### PR TITLE
osd: use fmt/ranges.h for printing ranges

### PR DESCRIPTION
**Pull request by tchaikov**
now that fmt/ranges.h provides a generic mechinary for printing
ranges. let's just use it.

the output format is changed a little bit when it comes to
set and map, but as long as it makes sense, and we don't use
a tool to parse the logging messages for extracting information
from the messages printed by fmt::format(), we should be safe.

Signed-off-by: Kefu Chai <tchaikov@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
hi Ronen, i was just skimming through https://hackingcpp.com/cpp/libs/fmt.html, and realized that we could simplify the code with the formatters defined in `<fmt/ranges.h>`!

File: src/include/types_fmt.h [hunk 1]
```diff
@@ -1,67 +0,0 @@
...
-
-  template <typename FormatContext>
-  auto format(const std::vector<Args...>& v, FormatContext& ctx)
-  {
-    return fmt::format_to(ctx.out(), "[{}]", fmt::join(v, ","));
-  }
-};
-
-template <class... Args>
-struct fmt::formatter<std::set<Args...>> {
```
Comment (rzarzynski) [inline src/include/types_fmt.h:59]: :heart:

**Review by rzarzynski — APPROVED** [review]

**Review by athanatos — APPROVED** [review]

**Comment by tchaikov** [comment]
changelog

- add missing `#include <fmt/ranges.h>` to `src/test/osd/scrubber_generators.cc`

**Comment by tchaikov** [comment]
jenkins test api

**Comment by ronen-fr** [comment]
Love this change. A lot was made possible when we've upgraded the fmtlib version.

**Pull request merged**